### PR TITLE
[BugFix] Using Objects.equals() to avoid NPE

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

com.naver.nid.cover.parser.coverage.exception.ParseException: java.nio.file.NoSuchFileException: /home/runner/_work/starrocks/starrocks/fe/fe-core/target/result

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 1]
```diff
@@ -624,6 +625,7 @@ public TGetTaskRunInfoResult getTaskRuns(TGetTasksParams params) throws TExcepti
         GlobalStateMgr globalStateMgr = GlobalStateMgr.getCurrentState();
         TaskManager taskManager = globalStateMgr.getTaskManager();
         List<TaskRunStatus> taskRunList = taskManager.showTaskRunStatus(null);
+        LOG.debug("task run status list: " + taskRunList);
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:628]: this debug should be removed

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 2]
```diff
@@ -580,6 +580,7 @@ public TGetTaskInfoResult getTasks(TGetTasksParams params) throws TException {
         GlobalStateMgr globalStateMgr = GlobalStateMgr.getCurrentState();
         TaskManager taskManager = globalStateMgr.getTaskManager();
         List<Task> taskList = taskManager.showTasks(null);
+        LOG.debug("task list: " + taskList);
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:583]: ditto

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java (hunk 1 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java]: done

**Pull request merged**